src/mcts/search.cc

```diff
@@ -898,6 +952,18 @@ SearchWorker::NodeToProcess SearchWorker::PickNodeToExtend(
             search_->remaining_playouts_ < best_node_n - child.GetN()) {
           continue;
         }
+        // If play certain win and don't search other
+        // moves at root. If search limit infinite continue searching other
+        // moves.
+        if (params_.GetCertaintyPropagation() &&
+            child.edge()->IsCertainWin()) {
```

Comment:
I think we might want this disabled for training, even in a scenario where we turn on certainty propagation on in training. Let certainty shape the policy through its confidence of Q - not by excluding visits to other children that might also turn out to be certain wins.

Reply:
Yes, this is something that warrants further discussion, as there are also other possibilities to use certainty information for training. The simplest (but maybe not best) way would be to just disable the "instant win play" as you propose. And if this has had its run in the wild and is tested well enough, it might be the way to go initially. But one could also, for example, stop the game as soon as a certain win is found and increase game generation rate or for example assign virtual visits to certain wins/draws and losses to maximize our current visit count of 800.
So if you feel that training on CP is something that will happen relatively soon, i would also disable this for training. If you feel we have some time before that happens, we could maybe come up with something better.